src/js/providers/providers.js

```diff
@@ -83,22 +83,34 @@ define([
             }));
         },
 
-        reorderProviders : function () {
-            // Remove the flash provider, and add it in front of the html5 provider
-            if (this.config.primary === 'flash') {
-                var flashIdx = _.indexOf(this.providers, _.findWhere(this.providers, {name: 'flash'}));
-                var flashProvider = this.providers.splice(flashIdx, 1)[0];
-                var html5Idx = _.indexOf(this.providers, _.findWhere(this.providers, {name: 'html5'}));
-                this.providers.splice(html5Idx, 0, flashProvider);
+        getPrimary: function(primaryOverride) {
+            var primary = primaryOverride || this.config.primary;
+
+            if (primary === 'html5' || primary === 'flash') {
+                return primary;
             }
+
+            return 'html5';
+        },
```

Comment:
I don't think we need an html5 primary setting. Primary is either set to 'flash' or it is not.

It's better that we limit it to this so that deprecation becomes simpler and we don't complicate what this setting does.